File: week2/Can-Yavuz/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  let maxProfit = 0;
  for (let left = 0, right = 1; right < prices.length; right++) {
    if (prices[left] < prices[right]) {
      const profit = prices[right] - prices[left];
      maxP = Math.max(maxProfit, profit);
    } else {
      left = right;
    }
  }
  return maxProfit;
};

File: week2/atakantekatan/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        LIST = []
        profit = 0
        for price in prices:
            if not LIST or price>LIST[0]:
                LIST.append(price)
            else:
                if len(LIST)>1:
                    profit = max(profit,max(LIST)-LIST[0])
                LIST = [price]
        return max(profit,max(LIST)-LIST[0])